curvefs/src/mds/topology/topology.cpp

```diff
@@ -351,18 +351,32 @@ TopoStatusCode TopologyImpl::UpdateMetaServerSpace(const MetaServerSpace &space,
         auto it = metaServerMap_.find(id);
         if (it != metaServerMap_.end()) {
             WriteLockGuard wlockMetaServer(it->second.GetRWLockRef());
-            diffCapacity = space.GetDiskCapacity() -
-                           it->second.GetMetaServerSpace().GetDiskCapacity();
+            diffCapacity = space.GetDiskThreshold() -
+                           it->second.GetMetaServerSpace().GetDiskThreshold();
             int64_t diffUsed = space.GetDiskUsed() -
                                it->second.GetMetaServerSpace().GetDiskUsed();
-            int64_t diffMemory =
+            int64_t diffDiskMinRequire =
+                space.GetDiskMinRequire() -
+                it->second.GetMetaServerSpace().GetDiskMinRequire();
+            int64_t diffMemoryThreshold =
+                space.GetMemoryThreshold() -
+                it->second.GetMetaServerSpace().GetMemoryThreshold();
+            int64_t diffMemoryUsed =
                 space.GetMemoryUsed() -
                 it->second.GetMetaServerSpace().GetMemoryUsed();
+            int64_t diffMemoryMinRequire =
+                space.GetMemoryMinRequire() -
+                it->second.GetMetaServerSpace().GetMemoryMinRequire();
             it->second.SetMetaServerSpace(space);
-            if (diffCapacity != 0 || diffUsed != 0 || diffMemory != 0) {
+            if (diffCapacity != 0 || diffUsed != 0 || diffDiskMinRequire != 0 ||
```

Comment:
`diffCapacity == 0` is not safe, what aboult `diffCapacity != 0`?

Reply:
diffCapacity is int64 type